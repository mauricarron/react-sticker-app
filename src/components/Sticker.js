import React from "react";

const Sticker = ({ id, title, url, setCopyState }) => {
  const handleClick = () => {
    const regex = /^(.*?).webp/;
    const copyURL = url.match(regex);
    console.log(copyURL);

    navigator.clipboard.writeText(url);

    setCopyState(true);
    setTimeout(() => {
      setCopyState(false);
    }, 2000);
  };

  return (
    <img
      key={id}
      alt={title}
      src={url}
      className="img-sticker"
      onClick={handleClick}
    />
  );
};

export default Sticker;
