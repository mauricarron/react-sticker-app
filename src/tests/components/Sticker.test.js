import { shallow } from "enzyme";
import Sticker from "../../components/Sticker";

describe("tests on SearchResults Component", () => {
  const id = "abc123";
  const title = "sticker title";
  const url = "https://mi.image.jpg";
  const setCopyState = () => {};

  const wrapper = shallow(
    <Sticker id={id} title={title} url={url} setCopyState={setCopyState} />
  );

  test("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should show props correctly", () => {
    const img = wrapper.find("img");

    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });
});
