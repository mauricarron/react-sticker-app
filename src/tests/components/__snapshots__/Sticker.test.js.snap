// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on SearchResults Component should match snapshot 1`] = `
<img
  alt="sticker title"
  className="img-sticker"
  key="abc123"
  onClick={[Function]}
  src="https://mi.image.jpg"
/>
`;
